Remotion/Validation/Globalization.UnitTests/TypeBasedValidationMessageFactoryTest.cs

```diff
@@ -1,32 +1,87 @@
+﻿using System.CodeDom;
+using Moq;
+using NUnit.Framework;
+using Remotion.Globalization;
+using Remotion.Reflection;
+using Remotion.ServiceLocation;
+using Remotion.Validation.Globalization.UnitTests.TestHelpers;
+using Remotion.Validation.Implementation;
+using Remotion.Validation.Validators;
 
 namespace Remotion.Validation.Globalization.UnitTests
 {
   [TestFixture]
   public class TypeBasedValidationMessageFactoryTest
   {
+    private TypeBasedValidationMessageFactory _factory;
+    private Mock<IPropertyInformation> _propertyStub;
+    private Mock<ITypeInformation> _objectStub;
+    private Mock<IMemberInformationGlobalizationService> _serviceStub;
+
+    [SetUp]
+    public void SetUp ()
+    {
+      _serviceStub = new Mock<IMemberInformationGlobalizationService>();
+      _factory = new TypeBasedValidationMessageFactory(_serviceStub.Object);
+
+      _propertyStub = new Mock<IPropertyInformation>();
+      _objectStub = new Mock<ITypeInformation>();
+    }
+
     [Test]
-    [Ignore("TODO RM-5906")]
     public void CreateValidationMessageForPropertyValidator_WithTypeNotHavingLocalizations_ReturnsNull ()
     {
+      var validator = new StubPropertyValidator();
+      var result = _factory.CreateValidationMessageForPropertyValidator(validator, _propertyStub.Object);
+
+      Assert.That(result, Is.Null);
     }
 
     [Test]
-    [Ignore("TODO RM-5906")]
     public void CreateValidationMessageForPropertyValidator_WithLocalizedType_ReturnsLocalizedValidationMessage ()
     {
+      var validator = new StubPropertyValidator();
+
+      var str = "LocalizedValidationMessage";
+      _serviceStub
+          .Setup(s => s.TryGetTypeDisplayName(
+              It.Is<ITypeInformation>(e => e is TypeAdapter && ((TypeAdapter)e).Type == typeof(StubPropertyValidator)),
+              It.IsAny<ITypeInformation>(),
+              out str))
+          .Returns(true);
+
+      var result = _factory.CreateValidationMessageForPropertyValidator(validator, _propertyStub.Object);
+
+      Assert.That(result, Is.InstanceOf<DelegateBasedValidationMessage>());
+      Assert.That(result.ToString(), Is.EqualTo("LocalizedValidationMessage"));
     }
 
     [Test]
-    [Ignore("TODO RM-5906")]
     public void CreateValidationMessageForObjectValidator_WithTypeNotHavingLocalizations_ReturnsNull ()
     {
+      var validator = new StubObjectValidator();
+      var result = _factory.CreateValidationMessageForObjectValidator(validator, _objectStub.Object);
+
+      Assert.That(result, Is.Null);
     }
 
     [Test]
-    [Ignore("TODO RM-5906")]
     public void CreateValidationMessageForObjectValidator_WithLocalizedType_ReturnsLocalizedValidationMessage ()
     {
+      var validator = new StubObjectValidator();
+
+      var str = "LocalizedValidationMessage";
+      _serviceStub
+          .Setup(s => s.TryGetTypeDisplayName(
+              It.Is<ITypeInformation>(e => e is TypeAdapter && ((TypeAdapter)e).Type == typeof(StubObjectValidator)),
```

Comment:
What does 'e' stand for? 

```suggestion
              It.Is<TypeAdapter>(e => e.Type == typeof(StubObjectValidator)),
```

I think this should work.  If so, please update the other tests, too.

Actually, there is an even better option: just use TypeAdapter.Get(typeof(StubObjectValidator)) for the first parameter and TypeAdapter.Get(_objectStub.Object.GetType()) for the second parameter. This way, we don';t need any `It.Is<..>` logic. You might need to capture those into variables before using them in the Setup call (not sure about this detail). The reason for the need to capture is that Moq might interpret the TypeAdapter.Get-call as part of the setup logic. So, first you can try if this works without using intermediary variables, and if not, introduce variables.

Reply:
@KuJo7 Since we went over this yesterday. I made a wrong assumption about `TypeAdapter`. While the equality uses reference equality the `TypeAdapter.Create` method caches instances so you will get back the same instance and equality works as expected. As mike has said this allows us to drop the complicated `It.Is` setup.

Side note: the method is called create so it is `TypeAdapter.Create(typeof(StubObjectValidator))` for both parameters as the type is used for both parameters for some reason.

Reply:
> Side note: the method is called create so it is TypeAdapter.Create(typeof(StubObjectValidator)) for both parameters as the type is used for both parameters for some reason.

This second bit, I want to post an update around noon.